fuse/gocryptfs/Portfile

```diff
@@ -0,0 +1,143 @@
+                        size    7808 \
+                    github.com/jacobsa/ogletest \
+                        lock    80d50a735a11 \
+                        rmd160  d839d0d084bc62e25d23d65cee882b427b537c00 \
+                        sha256  8e9eef43d47e4fc12b6e614bd6c97d102eabec107eabe72493763d45325501fd \
+                        size    24200 \
+                    github.com/jacobsa/oglemock \
+                        lock    e94d794d06ff \
+                        rmd160  cee16f043ba80e59483d3bfadf783077f8239110 \
+                        sha256  5474b9f5228d6eda866293feb9c07729a55195fabd2cd77cd1c30e5291830ad9 \
+                        size    35506 \
+                    github.com/jacobsa/oglematchers \
+                        lock    141901ea67cd \
+                        rmd160  5c5dd8416ee4a236632eaa4fc1a1e5ce737eee45 \
+                        sha256  4a45ac5d5b5c15af45761f15e3e14f4739b1cd79cd9493ddcd4744f65edc4b52 \
+                        size    43827 \
+                    github.com/jacobsa/crypto \
+                        lock    9f44e2d11115 \
+                        rmd160  0cbf3ba64ed7e4a65ad9e5083efe103deb2137f7 \
+                        sha256  8c230508b648a74dfecf86015965bd89c78b98a2e5988ee7ab538940db078390 \
+                        size    3653945 \
+                    github.com/hanwen/go-fuse \
+                        lock    ${gitversionfuse} \
+                        rmd160  60efe988c58e58c307bf86e3271ea7c419cd19eb \
+                        sha256  6fd20e10d9b30020f1cdee312c2906f85904894858bcd9248cf36e5912f2022f \
+                        size    204291 \
+                    github.com/davecgh/go-spew \
+                        lock    v1.1.0 \
+                        rmd160  0303eae19a01f38fe314921fd965e4d09b9ef3ad \
+                        sha256  810a597004388d68bb92d8aa612375419ba1080dd5fc2c66dd41b58f0ba4442c \
+                        size    42348
+
+set builddate       [exec date +"%Y-%m-%d"]
+set ldflags         "-X \"main.GitVersion=${version}\" -X \"main.GitVersionFuse=${gitversionfuse}\" -X \"main.BuildDate=${builddate}\""
+
+build.args          -ldflags="${ldflags}"
+
+# According to [1] there is no benefit to build gocryptfs with openssl on M1 Macs.
+# According to [2] on most modern CPUs openssl provides no speed benefits. I confirmed this by tests on
+# my Intel i7 Macbook Pro 2019. Therefore reducing the dependencies and defaulting to without_openssl.
+# [1]: https://github.com/rfjakob/gocryptfs/issues/556#issuecomment-848185514
+# [2]: https://github.com/rfjakob/gocryptfs/wiki/CPU-Benchmarks
+if {![variant_isset openssl]} {
+    default_variants    +without_openssl
+}
+
+variant openssl description {Build with openssl support} {
+    depends_lib-append  port:openssl
+}
+
+variant without_openssl description {Build without openssl support} {
+    build.env           CGO_ENABLED=0
+    build.args-append   -tags without_openssl
+}
```

Comment:
We should always avoid variants with prefixes such as `no_` or `without_`, as the inverse logic is confusing.

The `+openssl` variant only adds a dependency on a port. This implies that if `openssl` is already installed, it will also opportunistically be linked. This would add a hidden dependency that is not recorded in the registry and the user could simply uninstall openssl, which would break gocryptfs without warning. The missing dependency declaration can also cause problems when openssl gets ABI incompatible upgrades.

As the variant without openssl should be default, I suggest the following. Now either openssl is added explicitly with `+openssl`, or it will not be used. I also rewrote the dependency on openssl as `path:lib/libcrypto.dylib:openssl`, so it can be satisfied by other ports providing libcrypto.

```
variant openssl description {Build with openssl support} {
    depends_lib-append  path:lib/libcrypto.dylib:openssl
}

if {![variant_isset openssl]} {
    build.env           CGO_ENABLED=0
    build.args-append   -tags without_openssl</span>
}
````